iOS/ViewControllers/Courses/CourseProgressViewController.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  Created for xikolo-ios under MIT license.
+//  Copyright © HPI. All rights reserved.
+//
+
+import BrightFutures
+import Common
+import CoreData
+import SafariServices
+import UIKit
+
+class CourseProgressViewController: UITableViewController {
+
+    private var dataSource: CoreDataTableViewDataSource<CourseProgressViewController>!
+    var course: Course!
+
+    private lazy var courseProgress: CourseProgress? = {
+        return CoreDataHelper.viewContext.fetchSingle(CourseProgressHelper.FetchRequest.courseProgress(forCourse: course)).value
+    }()
+
+    @IBOutlet private weak var courseProgressView: CourseProgressView!
+
+    weak var scrollDelegate: CourseAreaScrollDelegate?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        self.addRefreshControl()
+
+        // setup table view data
+        let request = SectionProgressHelper.FetchRequest.sectionProgresses(forCourse: course)
+        let reuseIdentifier = R.reuseIdentifier.sectionProgressCell.identifier
+        let resultsController = CoreDataHelper.createResultsController(request, sectionNameKeyPath: nil)
+        self.dataSource = CoreDataTableViewDataSource(self.tableView,
+                                                      fetchedResultsController: resultsController,
+                                                      cellReuseIdentifier: reuseIdentifier,
+                                                      delegate: self)
+
+        self.refresh()
+
+        if let progress = self.courseProgress {
+            self.courseProgressView.configure(for: progress, showCourseTitle: self.course == nil)
+            self.tableView.resizeTableHeaderView()
+        }
```

Comment:
Here, we are running into a race condition. If on the initial load, the server hasn't responded then the course progress is not configured.
You need to perform two steps to resolve this issue:
- Add an else block that hides the table header. There is no sense in showing an non-configured course progress.
- `RefreshableViewController` has the method `didRefresh` which you can use to reconfigure (and show the course progress). Considering this, there should be no need for the member attribute `courseProgress` anymore as we have to fetch the course progress every time.